### ISSUE TITLE
Phusion Passenger error: listen EADDRINUSE: address already in use passenger

### ISSUE THREAD

**Issue by quantumwebco** [fastify/fastify#4278]
### Prerequisites

- [X] I have written a descriptive issue title
- [X] I have searched existing issues to ensure the issue has not already been raised

### Issue

When attempting to run a simple server with Phusion Passenger (in Plesk) I get the error ```listen EADDRINUSE: address already in use passenger```. I have followed any docs/fixes I can find, this stackoverflow https://stackoverflow.com/a/20645549/10628869 led me to https://www.phusionpassenger.com/library/indepth/nodejs/reverse_port_binding.html and everything seems set up correctly but I just get the error. 

It works fine with the example for express

Is Fastify not compatible with passenger? Or is there some setting I am missing to make it work?

This is my code, it works locally without passenger, I have also tried with async/await and other examples in the documentation
```
if (typeof (PhusionPassenger) != "undefined") {
    PhusionPassenger.configure({autoInstall: false});
}

const fastify = require("fastify")({logger: true});

fastify.get("/test", async (request, reply) => {
    reply.send({ hello: 'world' })
});

serverConfig = {};
if (typeof (PhusionPassenger) != "undefined") {
    serverConfig.path = "passenger";
}
else {
    serverConfig.port = 3000;
}
fastify.listen(serverConfig, function(err, address) {
    if (err) {
        fastify.log.error(err);
        process.exit(1);
    }
    // Server is now listening on ${address}
});
```

**Comment by mcollina** [comment]
I would recommend you open a ticket in PhusionPassenger.
I don't know why it's not working, however, I'm not a PhusionPassenger user, so I don't really know where to start.

**Comment by Eomm** [comment]
It should be this the answer
https://stackoverflow.com/questions/73563025/how-to-reverse-port-biding-phusion-passenger-and-fastify

**Comment by quantumwebco** [comment]
I've tried that but I still get the same issue

**Comment by Eomm** [comment]
Not joking: turn it off and on passengers
It could be a zombie instance: the error message is clear

**Comment by quantumwebco** [comment]
Tried that, rebooted the server etc. If I use ```fastify.listen("127.0.0.1")``` I get the error "type":"Error","message":"listen EADDRINUSE: address already in use 127.0.0.1","stack":"Error: listen EADDRINUSE: address already in use 127.0.0.1\n at Server.setupListenHandle

I think I just give up with Fastify. It seems like a good idea but every other framework just works out the box so I'll move to a framework that is a bit more useable and better supported

Thanks anyway. Good luck

**Comment by Eomm** [comment]
Your words are very tough I hope you understand them.
Moreover, your code does not show me that you read the documentation as your syntax uses the `host` option instead of `path` parameter as written in the SO answer.
Checkout the [Node.js core doc](https://nodejs.org/api/net.html#serverlistenoptions-callback) to more details.

> It seems like a good idea but every other framework just works out the box

Here your copy-paste fix: `fastify.listen({ path: 'passenger', host: '127.0.0.1' })`

**Comment by quantumwebco** [comment]
I apologise for any misunderstanding. I simply meant that Phusion and Plesk etc. have docs for and support other frameworks and the other frameworks I tried worked without issue. I had already tried your "copy-paste" suggestion and everything from SO questions and github issues etc. but to no avail. Fastify looks like a really good framework, it's a great idea, but unfortunately I ran out of time and had to use another solution that worked easily with my stack. I meant no disrespect and would love to use Fastify in the future on a stack that properly supports it. Thanks much for your time and support and good luck with life, the universe and everything :)
